libuavcan/include/libuavcan/time.hpp

```diff
@@ -14,153 +15,252 @@
 
 namespace libuavcan
 {
+namespace duration
+{
 /**
  * Protected base class for duration values.
  *
- * @tparam DurationType  The datatype returned when retrieving durations from
- *                       realizations of this base class. This type must be
- *                       exactly 8 bytes in size and must support certain concepts
- *                       implied by DurationBase.
+ * @tparam Type          The type of the derived class. This must be an empty type.
+ *                       All storage will be provided by this base class.
+ * @tparam USecT         The datatype returned when retrieving durations from
+ *                       realizations of this base class. This type must be signed.
  */
-template <typename DurationType>
-class DurationBase
+template <typename Type, typename USecT = std::int64_t>
+class Base
 {
-    std::int64_t usec_; /**< Internal storage of the duration value in microseconds. */
+    USecT usec_; /**< Internal storage of the duration value in microseconds. */
```

Comment:
```suggestion
    USecT usec_ = 0; /**< Internal storage of the duration value in microseconds. */
```
HIC++ 12.4.2, C++CG C.45

Reply:
Same comment as above: per c.21 will define all default operations.